internal/controller/grafanaorganization_controller.go

```diff
@@ -199,67 +308,40 @@ func (r GrafanaOrganizationReconciler) reconcileDelete(ctx context.Context, graf
 	return nil
 }
 
-// SetupWithManager sets up the controller with the Manager.
-func (r *GrafanaOrganizationReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	return ctrl.NewControllerManagedBy(mgr).
-		For(&v1alpha1.GrafanaOrganization{}).
-		// Watch for grafana pod's status changes
-		Watches(
-			&v1.Pod{},
-			handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, obj client.Object) []reconcile.Request {
-				k8sClient := mgr.GetClient()
-				var organizations v1alpha1.GrafanaOrganizationList
-
-				err := k8sClient.List(ctx, &organizations)
-				if err != nil {
-					log.FromContext(ctx).Error(err, "failed to list grafana organization CRs")
-					return []reconcile.Request{}
-				}
-
-				// Reconcile all grafana organizations when the grafana pod is recreated
-				requests := make([]reconcile.Request, 0, len(organizations.Items))
-				for _, organization := range organizations.Items {
-					requests = append(requests, reconcile.Request{
-						NamespacedName: types.NamespacedName{
-							Name: organization.Name,
-						},
-					})
-				}
-				return requests
-			}),
-			builder.WithPredicates(predicates.GrafanaPodRecreatedPredicate{}),
-		).
-		Complete(r)
-}
-
 // configureGrafana ensures the RBAC configuration is set in Grafana.
 func (r *GrafanaOrganizationReconciler) configureGrafana(ctx context.Context) error {
 	logger := log.FromContext(ctx)
 
-	organizations := v1alpha1.GrafanaOrganizationList{}
-	err := r.Client.List(ctx, &organizations)
+	organizationList := v1alpha1.GrafanaOrganizationList{}
+	err := r.Client.List(ctx, &organizationList)
 	if err != nil {
 		logger.Error(err, "failed to list grafana organizations.")
 		return errors.WithStack(err)
 	}
 
-	grafanaConfig := v1.ConfigMap{
+	grafanaConfig := &v1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "grafana-user-values",
 			Namespace: "giantswarm",
 		},
 	}
 
-	_, err = controllerutil.CreateOrPatch(ctx, r.Client, &grafanaConfig, func() error {
-		config, err := templating.GenerateGrafanaConfiguration(organizations.Items)
+	_, err = controllerutil.CreateOrPatch(ctx, r.Client, grafanaConfig, func() error {
+		organizations := organizationList.Items
+		// We always sort the organizations to ensure the order is deterministic and the configmap is stable
```

Comment:
This is needed because this causes unecessary restarts of grafana otherwise

Reply:
Maybe add this as a comment

Reply:
I did, thé configmap must bé stable was clear to me 😅

Reply:
```suggestion
		// We always sort the organizations to ensure the order is deterministic and the configmap is stable
		// in order to prevent grafana to restarts.
```